storage/range.go

```diff
@@ -765,6 +765,7 @@ func (r *Range) processRaftCommand(idKey cmdIDKey, index uint64, raftCmd proto.I
 
 // applyRaftCommand applies a raft command from the replicated log to
 // the underlying state machine (i.e. the engine).
+// The caller needs to hold the range lock.
```

Comment:
Unfortunately not; I wish go had something like google's c++ Mutex::AssertHeld(). The best you can do is rely on the race detector (which kind of worked in this case, but the race didn't manifest reliably enough).